crates/provider-keyvalue-redis/src/lib.rs

```diff
@@ -141,6 +163,54 @@ impl KvRedisProvider {
         Ok(conn.clone())
     }
 
+    async fn invoke_on_set(&self, target: impl AsRef<str>, bucket: &str, key: &str, value: &str) {
+        let mut cx: async_nats::HeaderMap = async_nats::HeaderMap::new();
+        for (k, v) in
+        wasmcloud_provider_sdk::wasmcloud_tracing::context::TraceContextInjector::default_with_span(
+        )
+        .iter()
+    {
+        cx.insert(k.as_str(), v.as_str())
+    }
```

Comment:
@vados-cosmonic there's a macro that does this right? Is it https://docs.rs/wasmcloud-provider-sdk/0.11.0/wasmcloud_provider_sdk/macro.propagate_trace_for_ctx.html?

Reply:
Yup, @Aditya1404Sal you should be able to use `propagate_trace_for_ctx` -- here's [an example](https://github.com/wasmCloud/wasmCloud/blob/6fbf413df073f79b6d52a857b722f0cd7003a07e/crates/provider-keyvalue-nats/src/lib.rs#L328)